assets/sass/protocol/components/_button.scss

```diff
@@ -8,7 +8,8 @@
 @mixin default-states {
     &:focus {
         border-color: $button-border-color-focus;
```

Comment:
**nit (non-blocking, pre-existing)**: we have redundant border-color rules on neutral buttons

**suggestion**: while we're here, let's remove the redundant rules under `mzp-t-neutral` styles. Neutral buttons already inherit default state styles from `mzp-c-button`. We should only override the border-color property if it needs to be different.
<img width="539" alt="neutral" src="https://user-images.githubusercontent.com/19650432/227174873-7a7a93b1-9e97-451b-bc32-b3e809b831b2.png">

Reply:
The neutral states have their own state styles different from the primary and secondary buttons, so if I were to remove their styles, we'd be removing their state styles entirely. The other button styles are able to inherit default styles because the default/product styles mixins are being passed into the respective classes using the `@include` rule. 

Also, if I do remove the neutral states and pass along the default-styles mixin, I'd essentially be getting rid of  the neutral styles we have embedded. 🤕

Reply:
Sorry, I think my suggestion was badly worded. The only redundant rules I would want to remove is the Neutral button :focus border-color on L222 & L244. They're not actually using a custom "neutral" focus color, but sharing the `$button-border-color-focus` variable.

The more I look at it, actually, the more I find it strange we re-declare the `default-states` in `mzp-t-button-secondary` too
<img width="297" alt="redundant" src="https://user-images.githubusercontent.com/19650432/227975098-c3bf1ccc-ab66-4d31-a668-2275b4d8f0f6.png">
<img width="292" alt="inherited" src="https://user-images.githubusercontent.com/19650432/227975102-562c6fbf-4707-4365-8b3e-3efa2ded4bf9.png">

I'm going to create a separate issue to address this. Let's merge these focus states!

Reply:
In the end, I couldn't remove the secondary rule because it would require use of `:not(:hover, :focus, :active)` selector on the theme styles, which isn't supported in IE11. Leaving this in draft form: https://github.com/mozilla/protocol/pull/866
You can incorporate it here or merge as-is and I'll update the changelog on the draft PR

Reply:
hahaha, thanks!